server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -207,6 +210,15 @@ private RolloverResult rolloverAlias(ClusterState currentState, IndexAbstraction
     private RolloverResult rolloverDataStream(ClusterState currentState, IndexAbstraction.DataStream dataStream, String dataStreamName,
                                               CreateIndexRequest createIndexRequest, List<Condition<?>> metConditions,
                                               boolean silent, boolean onlyValidate) throws Exception {
+
+        if (SnapshotsService.snapshottingDataStreams(currentState, Collections.singleton(dataStream.getName())).isEmpty() == false) {
+            throw new SnapshotInProgressException(
```

Comment:
If `SnapshotInProgressException` is thrown and a data stream is managed by ilm then ilm will retry, so that looks good to me.